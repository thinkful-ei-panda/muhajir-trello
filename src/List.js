import React from 'react';
import Card from './Card';


function List(props) {
  return (
    <section className="List">
      <header className="List-header">
      <h2>{props.header}</h2>
      </header>
      <div className="ListCards">
          <button type="button" onClick={() => props.handleAdd(props.listId)}>Add Random Card</button>
          {props.card.map(card => 
            <Card key={card.id} id={card.id} title={card.title} content={card.content} handleDelete={props.handleDelete}/>
            )}
          
      </div>
    </section>
  )
}

export default List