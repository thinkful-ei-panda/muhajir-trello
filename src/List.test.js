import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import List from './List';

describe('<List />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <List id="id" header="header" cardIds={['a','b']} allCards={
        {
          a: { id: 'a', title: 'First card', content: 'lorem ipsum' },
          b: { id: 'b', title: 'Second card', content: 'lorem ipsum' }
        }
      }
      />,
      div
    );
    
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    const tree = renderer.create(<List id="id" header="header" cardIds={['a','b']} allCards={
      {
        a: { id: 'a', title: 'First card', content: 'lorem ipsum' },
        b: { id: 'b', title: 'Second card', content: 'lorem ipsum' }
      }
    }/>).toJSON();

    expect(tree).toMatchSnapshot();
  })
});