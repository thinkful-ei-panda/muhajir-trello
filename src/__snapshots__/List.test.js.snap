// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header
    </h2>
  </header>
  <div
    className="ListCards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
